test/dotnet-new.Tests/GivenThatIWantANewApp.cs

```diff
@@ -14,58 +14,65 @@ namespace Microsoft.DotNet.New.Tests
+                .WithWorkingDirectory(testDirectory)
                 .ExecuteWithCapturedOutput("new");
 
-            DateTime actualState = Directory.GetLastWriteTime(rootPath);
+            var actualState = testDirectory.LastWriteTime;
 
             Assert.Equal(expectedState, actualState);
 
-            result.Should().Fail()
-                  .And.HaveStdErr();
+            result
+                .Should().Fail()
+                     .And.HaveStdErr();
         }
  
         [Fact] 
         public void RestoreDoesNotUseAnyCliProducedPackagesOnItsTemplates() 
         { 
             var cSharpTemplates = new [] { "Console", "Lib", "Web", "Mstest", "XUnittest" }; 
  
-            var rootPath = TestAssetsManager.CreateTestDirectory().Path; 
-            var packagesDirectory = Path.Combine(rootPath, "packages"); 
+            var testDirectory = TestAssets.CreateTestDirectory();
+
+            var packagesDirectory = testDirectory.GetDirectory("packages");
  
-            foreach (var cSharpTemplate in cSharpTemplates) 
-            { 
-                var projectFolder = Path.Combine(rootPath, cSharpTemplate); 
-                Directory.CreateDirectory(projectFolder); 
-                CreateAndRestoreNewProject(cSharpTemplate, projectFolder, packagesDirectory); 
+            foreach (var cSharpTemplate in cSharpTemplates)
+            {
+                var projectDirectory = testDirectory.GetDirectory(cSharpTemplate);
+
+                projectDirectory.Create();
+
+                CreateAndRestoreNewProject(cSharpTemplate, projectDirectory, packagesDirectory); 
             } 
  
-            Directory.EnumerateFiles(packagesDirectory, $"*.nupkg", SearchOption.AllDirectories) 
-                .Should().NotContain(p => p.Contains("Microsoft.DotNet.Cli.Utils")); 
+            packagesDirectory
+                .Should().NotHaveFilesMatching($"Microsoft.DotNet.Cli.Utils.*.nupkg", SearchOption.AllDirectories); 
         } 
  
         private void CreateAndRestoreNewProject( 
             string projectType, 
-            string projectFolder, 
-            string packagesDirectory) 
+            DirectoryInfo projectDirectory, 
+            DirectoryInfo packagesDirectory) 
         { 
-            new TestCommand("dotnet") { WorkingDirectory = projectFolder } 
-                .Execute($"new --type {projectType}") 
-                .Should().Pass(); 
+            new TestCommand("dotnet")
+                .WithWorkingDirectory(projectDirectory)
+                .Execute($"new --type {projectType}")
+                .Should().Pass();
  
             new RestoreCommand() 
-                .WithWorkingDirectory(projectFolder) 
-                .Execute($"--packages {packagesDirectory} /p:SkipInvalidConfigurations=true") 
-                .Should().Pass(); 
+                .WithWorkingDirectory(projectDirectory)
+                .Execute($"--packages {packagesDirectory.FullName} /p:SkipInvalidConfigurations=true")
```

Comment:
this is currently not biting us but it might be worth considering making Execute take `string[]` - i already had some tiny issues with path escaping. If directory has spaces this might fail i.e.. Also if directory has backslash in the end even this will fail:
```
-- packages \"{path}\"
```
and the error will look really obscure IMO